src/reanimated2/Hooks.js

```diff
@@ -345,14 +346,51 @@ export function useDerivedValue(processor, dependencies = undefined) {
   return sharedValue;
 }
 
-export function useAnimatedGestureHandler(handlers) {
+// builds one big hash from multiple worklets' hashes
+function buildWorkletsHash(handlers) {
+  return Object.keys(handlers).reduce(
+    (previousValue, key) =>
+      previousValue === null
+        ? handlers[key].__workletHash
+        : previousValue.toString() + handlers[key].__workletHash.toString(),
+    null
+  );
+}
+
+export function useAnimatedGestureHandler(handlers, dependencies) {
   const initRef = useRef(null);
   if (initRef.current === null) {
     initRef.current = {
       context: makeRemote({}),
+      savedDependencies: [],
     };
   }
-  const { context } = initRef.current;
+  const { context, savedDependencies } = initRef.current;
+  let dependenciesDiffer = false;
+
+  if (dependencies === undefined) {
+    dependencies = Object.keys(handlers).map((handlerKey) => {
+      const handler = handlers[handlerKey];
+      return {
+        workletHash: handler.__workletHash,
+        closure: handler._closure,
+      };
+    });
+  } else {
+    dependencies.push(buildWorkletsHash(handlers));
+  }
+
+  if (dependencies.length !== savedDependencies.length) {
+    dependenciesDiffer = true;
+  } else {
+    for (let i = 0; i < dependencies.length; ++i) {
+      if (dependencies[i] !== savedDependencies[i]) {
```

Comment:
this is probably not a good solution for checking if dependencies changed

Reply:
I would use at least Object.is for that. React itself is using similar thing afaik

Reply:
Yes, I just wanted to mark it here so I can see that it should be done in a different way